lib/senkyoshi/canvas_course.rb

```diff
@@ -63,9 +62,49 @@ def self.from_metadata(metadata, blackboard_export = nil)
     # Creates a canvas assignment from a scorm package that has already been
     # uploaded to a scorm manager
     ##
-    def create_scorm_assignment(scorm_package, course_id)
-      url = Senkyoshi.scorm_launch_url +
-        "?course_id=#{scorm_package['package_id']}"
+    def create_scorm_assignment(scorm_package, course_id, local)
+      if local
+        _create_scorm_assignment_local(scorm_package)
+      else
+        _create_scorm_assignment_external(scorm_package, course_id)
+      end
+    end
+
+    ##
+    # Assembles the launch url with the course_id
+    ##
+    def _scorm_launch_url(package_id)
+      "#{Senkyoshi.configuration.scorm_launch_url}?course_id=#{package_id}"
+    end
+
+    ##
+    # Creates a scorm assignment from a Canvas course object
+    ##
+    def _create_scorm_assignment_local(scorm_package)
```

Comment:
I saw line 67, I just couldn't figure out where in the world the 'if local' branch was getting run. That makes sense though how ha.